loadbalancer/zk/publisher_test.go

```diff
@@ -43,13 +44,16 @@ func TestBadFactory(t *testing.T) {
 	}
 	defer p.Stop()
 
+	// instance1 came online
+	client.AddService(path+"/instance1", "zookeeper_node_data")
+	time.Sleep(1 * time.Millisecond)
```

Comment:
No I'm not, so we have two options:

1: longer timeout in main test goroutine to yield long enough to have publisher goroutine handle the update. this will slow down the unit test.
2: loop until expected value has been propagated or higher set timeout occurs (error). 

An unfortunate side effect from having to test asynchronous updates.